### ISSUE TITLE
[5.x]: All asset relationships are gone when upgrading from 4 to 5

### ISSUE THREAD

**Issue by jaspertandy** [craftcms/cms#15928]
### Description

I've just tried to upgrade a site locally from Craft 4 to Craft 5 and all images are now missing. They're still there in the assets folder but they're not assigned to entries any more.

### Steps to reproduce

1. I have a specific database and setup I can reproduce this on, but it's not something I can share and I can't see why it's happening. I am happy to demo this to a developer and it's on a Craft pro license so hopefully we can work something out?

### Expected behavior

When I run `php craft up` my assigned images are retained

### Actual behavior

Images aren't retained

### Craft CMS version

5.4.8

### PHP version

8.2.24

### Installed plugins and versions

-

**Comment by brandonkelly** [comment]
Strange, the Craft 5 migration doesn’t touch the `relations` table, where all relations are stored.

Does that table still have all its rows in it?

**Comment by jaspertandy** [comment]
So I actually don't seem to have anything in this table. I'm probably not using the right nomenclature.

```
select e.*
from elements e 
inner join relations r on r.targetId = e.id
where e.type = 'craft\elements\Asset'
```

Where would this association be stored? It's just a normal asset field with an asset selected.

![Image](https://github.com/user-attachments/assets/c960415c-09d2-4ccc-921e-656f8d790e75)

**Comment by brandonkelly** [comment]
Yeah those relations are stored in `relations`. However you need to escape the slashes in the `e.type` condition:

```sql
select e.*
from elements e 
inner join relations r on r.targetId = e.id
where e.type = 'craft\\elements\\Asset'
```

**Comment by jaspertandy** [comment]
Ah damn it I was so close. Well the offending upgrade went through successfully after I noticed that `php craft up` ran out of RAM. It didn't seem to be doing anything at the time but I wondered if it might've caused issues doing things it wasn't logging to stdout so I increased the amount of RAM the task was allowed and it's gone through fine now. Unfortunately can't say what specifically caused it but this is a site with a lot of sections, fields, and entries so I expect lacking RAM could've upset it.

Thanks, and sorry for the red herring.
